### DISCUSSION TITLE
Multiple audio outs?

### DISCUSSION THREAD

**Discussion by armandhammer** [probonopd/MiniDexed#113] [Q&A] [answered]
Is there any way to add individual audio outs for each of the 8 generators, like on the TX802?

**Comment by probonopd** [comment, marked as answer]
Hello @bengarland, I don't think this is possible to attach multiple audio outs to one Raspberry Pi other than via USB (which is not supported in MiniDexed). Hence I believe you'd need 4 Raspberry Pis and 4 DACs, each doing 2 channels, although for this use case RPi Zero 2 W boards might be sufficient. On each RPi, you would need to set one TG stereo panned all the way to the left and one all the way to the right channel, and disable reverb.

__Update:__ There might be a way to do this: https://robertsonics.com/2021/11/17/multi-channel-audio-for-the-raspberry-pi-part-1/ but it would require a volunteer to do the work.

**Reply by probonopd** [reply]
According to https://forums.adafruit.com/viewtopic.php?f=8&p=568198#p710328,

> There are more advanced versions that use more signals to control more than two channels, but the RasPi's I2S bus doesn't support them. You only get two channels of I2S data on a RasPi.

**Reply by randyandrade** [reply]
> Hello @bengarland, I don't think this is possible to attach multiple audio outs to one Raspberry Pi other than via USB (which is not supported in MiniDexed). Hence I believe you'd need 4 Raspberry Pis and 4 DACs, each doing 2 channels, although for this use case RPi Zero 2 W boards might be sufficient. On each RPi, you would need to set one TG stereo panned all the way to the left and one all the way to the right channel, and disable reverb.

Is there any plans to add support for class-compliant USB-audio devices? I would really love to be able to use some balanced outputs with this project, and I haven't personally seen a hat which supports that. (or if you know of one, please let me know, I'd be curious about it)

I guess, I'm more curious if the USB Audio thing is a "hard no" as in definitely never happen, or a "soft no" as in, not currently supported, but  could possibly happen at some point.

**Reply by probonopd** [reply]
One should never say "never", but https://github.com/rsta2/circle/issues/222 suggests that is unlikely to happen anytime soon.

I don't know about balanced outputs.

**Reply by probonopd** [reply]
__Update:__ There might be a way to do this: https://robertsonics.com/2021/11/17/multi-channel-audio-for-the-raspberry-pi-part-1/ but it would require a volunteer to do the work.

**Reply by probonopd** [reply]
Circle 45 now has support for USB audio devices on Raspberry Pi 4, 400, and CM4.

https://github.com/rsta2/circle/releases/tag/Step45

Possibly it could be made work using that?
